src/main/scala/me/rexim/morganey/autocompletion/extractors/CommandWithArg.scala

```diff
@@ -0,0 +1,15 @@
+package me.rexim.morganey.autocompletion.extractors
+
+import me.rexim.morganey.Commands._
+
+class CommandWithArg(commands: Map[String, Command]) {
+  def unapply(line: String): Option[(String, String)] = {
+    val potentialCommand = parseCommand(line)
+    potentialCommand flatMap { case (p, arg) =>
+      commands.values find {
+        case StringCommand(_, _)  => false
+        case TermCommand(name, _) => name == p
+      } map (_.name -> arg)
+    }
+  }
+}
```

Comment:
But this answers only why we converted object into a class. The reason why we relocated it after that is because it's a command extractor and we have a [special package](https://github.com/morganey-lang/Morganey/tree/3802a283ef2ef35ba703988e2cc7a52c2bece313/src/main/scala/me/rexim/morganey/autocompletion/extractors) for them now